src/VisualStudio/Core/Def/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1464,10 +1420,9 @@ protected override void Dispose(bool finalize)
             _textBufferFactoryService.TextBufferCreated -= AddTextBufferCloneServiceToBuffer;
             _projectionBufferFactoryService.ProjectionBufferCreated -= AddTextBufferCloneServiceToBuffer;
 
-            if (_lazyExternalErrorDiagnosticUpdateSource.IsValueCreated)
-            {
-                _lazyExternalErrorDiagnosticUpdateSource.Value.Dispose();
-            }
+            // Make sure we unsubscribe this, or otherwise this will cause a leak in unit tests since the UIContext for SolutionClosing is a static that is shared
+            // across all tests.
+            _solutionClosingContext?.UIContextChanged -= SolutionClosingContext_UIContextChanged;
```

Comment:
not sure i entirely unnderstand why this is then ok.  but i trust you.

Reply:
The whole 'there is something static somewhere that is leaking, but this is actually an instance member that we're hooking up to'.